sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/CodeGeneratorWithInterpretedFallback.scala

```diff
@@ -63,7 +49,10 @@ abstract class CodeGeneratorWithInterpretedFallback[IN, OUT] {
         try {
           createCodeGeneratedObject(in)
         } catch {
-          case CodegenError(_) => createInterpretedObject(in)
+          case _: Exception =>
+            // We should have already seen the error message in `CodeGenerator`
+            logWarning("Expr codegen error and falling back to interpreter mode")
```

Comment:
Do we need to include the exception type we captured in the message?

Reply:
We could; although the detail exception type/message/stack trace info would already be printed in the preceding logging from the `CodeGenerator` so I wouldn't push strongly for adding the exception type here.
(It certainly is easier to do data science on just single lines of logs instead of having to search across a few lines for context, so I do see the benefits of adding the type information albeit redundant. I'll leave it up to you guys @maropu @viirya @cloud-fan @gatorsmile )

Reply:
+ 1 to keep the current message. `CodeGenerator` already has printed many infos for errors.